pkg/router/controller/factory/factory.go

```diff
@@ -35,10 +36,11 @@ type RouterControllerFactory struct {
 }
 
 // NewDefaultRouterControllerFactory initializes a default router controller factory.
-func NewDefaultRouterControllerFactory(oc osclient.RoutesNamespacer, kc kclient.EndpointsNamespacer) *RouterControllerFactory {
+func NewDefaultRouterControllerFactory(oc osclient.RoutesNamespacer, kc kclient.Interface) *RouterControllerFactory {
 	return &RouterControllerFactory{
 		KClient:        kc,
 		OSClient:       oc,
+		NClient:        kc,
```

Comment:
Yeah, if its same as kc - we should cleanup to just use 1 `?Client`. Maybe cleanup in follow on work. I know changing it would touch a lot more code in this PR.